python/ray/experimental/multiprocessing/pool.py

```diff
@@ -0,0 +1,488 @@
+        # modify the global namespace (e.g., import packages or set globals)
+        # due to a limitation in cloudpickle.
+        return (PoolActor.remote(self._initializer, self._initargs), 0)
+
+    # Batch should be a list of tuples: (args, kwargs).
+    def _run_batch(self, actor_index, func, batch):
+        actor, count = self._actor_pool[actor_index]
+        object_id = actor.run_batch.remote(func, batch)
+        count += 1
+        if count == self._maxtasksperchild:
+            self._stop_actor(actor)
+            actor, count = self._new_actor_entry()
+        self._actor_pool[actor_index] = (actor, count)
+        return object_id
+
+    def apply(self, func, args=None, kwargs=None):
+        return self.apply_async(func, args, kwargs).get()
+
+    def apply_async(self,
+                    func,
+                    args=None,
+                    kwargs=None,
+                    callback=None,
+                    error_callback=None):
+        self._check_running()
+        random_actor_index = random.randrange(len(self._actor_pool))
+        object_id = self._run_batch(random_actor_index, func, [(args, kwargs)])
+        return AsyncResult(
+            [object_id], callback, error_callback, single_result=True)
+
+    def map(self, func, iterable, chunksize=None):
+        self._check_running()
+        return self._map_async(
+            func, iterable, chunksize=chunksize, unpack_args=False).get()
+
+    def map_async(self,
+                  func,
+                  iterable,
+                  chunksize=None,
+                  callback=None,
+                  error_callback=None):
+        self._check_running()
+        return self._map_async(
+            func,
+            iterable,
+            chunksize=chunksize,
+            unpack_args=False,
+            callback=callback,
+            error_callback=error_callback)
+
+    def starmap(self, func, iterable, chunksize=None):
+        self._check_running()
+        return self._map_async(
+            func, iterable, chunksize=chunksize, unpack_args=True).get()
+
+    def starmap_async(self, func, iterable, callback=None,
+                      error_callback=None):
+        self._check_running()
+        return self._map_async(
+            func,
+            iterable,
+            unpack_args=True,
+            callback=callback,
+            error_callback=error_callback)
+
+    def _map_async(self,
+                   func,
+                   iterable,
+                   chunksize=None,
+                   unpack_args=False,
+                   callback=None,
+                   error_callback=None):
+        object_ids = self._chunk_and_run(
+            func, iterable, chunksize=chunksize, unpack_args=unpack_args)
+        return AsyncResult(object_ids, callback, error_callback)
+
+    def _calculate_chunksize(self, iterable):
+        chunksize, extra = divmod(len(iterable), len(self._actor_pool) * 4)
```

Comment:
Factor out and document magic number `4`?

Reply:
Was this resolved? It seems to say `4` at the moment, not sure if there are pending changes.